lenskit/lenskit/algorithms/basic.py

```diff
@@ -80,6 +84,55 @@ def __str__(self):
         return "PopScore({})".format(self.score_method)
 
 
+class TimeBoundedPopScore(PopScore):
+    """
+    Score items by their time-bounded popularity, i.e., the popularity in the
+    most recent `time_window` period.  Use with :py:class:`TopN` to get a
+    most-popular-recent-items recommender.
+
+    Args:
+        time_window(datetime.timedelta):
+            The time window for computing popularity scores.
+        score_type(str):
+            The method for computing popularity scores.  Can be one of the following:
+
+            - ``'quantile'`` (the default)
+            - ``'rank'``
+            - ``'count'``
+
+    Attributes:
+        item_scores_(pandas.Series):
+            Time-bounded item popularity scores.
+    """
+
+    def __init__(self, time_window: timedelta, score_method="quantile"):
+        super().__init__(score_method)
+
+        self.time_window = time_window
+        self.score_method = score_method
+
+    @override
+    def fit(self, data: Dataset, **kwargs):
+        _logger.info("counting time-bounded item popularity")
+
+        log = data.interaction_log("numpy")
+
+        counts = np.zeros(data.item_count, dtype=np.int32)
+        start_timestamp = (datetime.now() - self.time_window).timestamp()
+        item_nums = log.item_nums[log.timestamps is not None and log.timestamps > start_timestamp]
```

Comment:
I'm not sure what `log.timestamps is not None` will do here when it's actually None (we'll have a single value of `False` as the index). Better to put the `none` check outside of the indexing — if timestamps is `None`, then issue a warning (because time-bounded popularity without timestamps is meaningless) and fall back to the unbounded popularity.